File: FS/3.0/broker.cc
#include "lib/fs.h"
#include "lib/json.hpp"

using json = nlohmann::json;

int main() {
  cout << "This is the Broker\n" << endl;
  context ctx;
  socket b(ctx, socket_type::rep);

  cout << "Binding socket to tcp port 6666\n";
  b.bind("tcp://*:6666");

  return 0;
}

File: zpotify/broker.py
#!/usr/bin/python

import zmq
import ffile


def main():
    context = zmq.Context()
    socket = context.socket(zmq.REP)
    socket.bind("tcp://*:5555")

    while True:
        print("Broker waiting...")
        message = socket.recv()
        print("Received request: %s" % message)

        #  Send reply back to client
        socket.send(b"World")


if __name__ == '__main__':
    main()
